src/cpp/session/modules/data/SessionData.cpp

```diff
@@ -25,22 +27,168 @@ using namespace rstudio::core ;
 
 namespace rstudio {
 namespace session {
-namespace modules { 
+namespace modules {
 namespace data {
 
+class AsyncDataPreviewRProcess : public async_r::AsyncRProcess
+{
+public:
+   static boost::shared_ptr<AsyncDataPreviewRProcess> create(
+           const json::JsonRpcRequest& request,
+           const json::JsonRpcFunctionContinuation& continuation)
+   {
+      boost::shared_ptr<AsyncDataPreviewRProcess> pDataPreview(
+                  new AsyncDataPreviewRProcess(request, continuation));
+      pDataPreview->start();
+      return pDataPreview;
+   }
+
+   void terminateProcess()
+   {
+      async_r::AsyncRProcess::terminate();
+   }
+
+private:
+   AsyncDataPreviewRProcess(
+           const json::JsonRpcRequest& request,
+           const json::JsonRpcFunctionContinuation& continuation) :
+       continuation_(continuation),
+       request_(request)
+   {
+   }
+
+   FilePath pathFromModulesSource(std::string sourceFile)
+   {
+      FilePath modulesPath = session::options().modulesRSourcePath();
+      FilePath srcPath = modulesPath.complete(sourceFile);
+
+      return srcPath;
+   }
+
+   FilePath pathFromSource(std::string sourceFile)
+   {
+      FilePath sourcesPath = session::options().coreRSourcePath();
+      FilePath srcPath = sourcesPath.complete(sourceFile);
+
+      return srcPath;
+   }
+
+   void start()
+   {
+      std::ostringstream jsonStream;
+      json::Value requestValue = request_.params;
+      json::write(requestValue, jsonStream);
+      std::string jsonData = jsonStream.str();
+
+      std::string cmd = ".rs.preview_data_import_async('" + jsonData + "')";
+
+      std::vector<core::FilePath> sources;
+      sources.push_back(pathFromSource("Tools.R"));
+      sources.push_back(pathFromModulesSource("ModuleTools.R"));
+      sources.push_back(pathFromModulesSource("SessionDataViewer.R"));
+      sources.push_back(pathFromModulesSource("SessionDataImportV2.R"));
+
+      async_r::AsyncRProcess::start(cmd.c_str(), FilePath(), async_r::R_PROCESS_VANILLA, sources);
```

Comment:
It looks like you can get access to Tools.R and ModuleTools.R by passing the `R_PROCESS_AUGMENTED` flag. @kevinushey I noticed that this flag also includes SessionCodeTools.R and SessionRCompletions.R. Perhaps those should be removed and specified explicitly as extra source files within SessionAsyncPackageInformation